js/src/viewer/manifestsPanel.js

```diff
@@ -129,15 +239,310 @@
            _this.hide();
         },
 
+        // Handler for when manifest data is loaded for the first time
         onManifestReceived: function(event, newManifest) {
           var _this = this;
-          _this.manifestListItems.push(new $.ManifestListItem({ 
-            manifest: newManifest, 
-            resultsWidth: _this.resultsWidth, 
-            state: _this.state,
-            eventEmitter: _this.eventEmitter,
-            appendTo: _this.manifestListElement }));
-          _this.element.find('#manifest-search').keyup();
+          // Show a manifest list item only if the currently selected "folder" expects it
+          if (_this.expectedThings.indexOf(newManifest.uri) != -1) {
+            _this.manifestListItems.push(new $.ManifestListItem({ 
+              manifest: newManifest, 
+              resultsWidth: _this.resultsWidth, 
+              state: _this.state,
+              eventEmitter: _this.eventEmitter,
+              forcedIndex: _this.expectedThings.indexOf(newManifest.uri),
+              appendTo: _this.manifestListElement }));
+            _this.element.find('#manifest-search').keyup();
+          }
+        },
+        
+        // Handler for when collection data is loaded for the first time
```

Comment:
I labelled these "first time" because I've only observed them run once when the JSON data is first loaded. On subsequent attempts to load, current Mirador components fetch from the state and bypass the events.